metrics/service.go

```diff
@@ -0,0 +1,139 @@
+package metrics
+
+import (
+	"log/slog"
+	"time"
+
+	"github.com/dansimau/hal/store"
+	"gorm.io/gorm"
+)
+
+// Service handles metrics collection and pruning
+type Service struct {
+	db             *gorm.DB
+	batchSize      int
+	metricsBuffer  []store.Metric
+	bufferChan     chan store.Metric
+	pruneInterval  time.Duration
+	retentionTime  time.Duration
+	stopChan       chan struct{}
+}
+
+// NewService creates a new metrics service
+func NewService(db *gorm.DB) *Service {
+	return &Service{
+		db:            db,
+		batchSize:     100,
+		metricsBuffer: make([]store.Metric, 0, 100),
+		bufferChan:    make(chan store.Metric, 1000), // Buffer up to 1000 metrics
```

Comment:
If we buffer up to 1000 metrics, this means we will lose them if the process dies, which is bad. I think this is a premature optimisation. Let's focus on writing to sqlite immediately and let the WAL handle flushing.